zebra/if_netlink.c

```diff
@@ -1330,6 +1334,21 @@ int netlink_link_change(struct nlmsghdr *h, ns_id_t ns_id, int startup)
 							"Intf %s(%u) PTM up, notifying clients",
 							name, ifp->ifindex);
 					zebra_interface_up_update(ifp);
+
+					/* Update EVPN VNI when SVI MAC change
+					 */
+					if (IS_ZEBRA_IF_VLAN(ifp) &&
+					    memcmp(old_hw_addr, ifp->hw_addr,
```

Comment:
If we had received events in the below sequence
1) An update with MAC address change and interface is down
2) Followed by An update with interface is UP.

In this sequence, below code misses to send a SVI UP event. Right ? Is this expected ?

Reply:
SVI UP is already called as part of interface UP event, The missing scenario was svi is already up and mac address change did not trigger to send EVPN updates.

Reply:
1) For an interface down event and also with a MAC address change, when function netlink_link_change()  is called, as per the new code, zebra_vxlan_svi_up() function will not be called as interface is DOWN.
2) Now for an interface UP event following the above event, when function netlink_link_change() is called, zebra_vxlan_svi_up() is not called as there is no change in MAC this time.

This is the scenario I’m pointing out. If you know another work flow where zebra_vxlan_svi_up() is actually called in the above scenario or if I am missing something here, please let me know. Other changes are fine and I can approve and merge the PR accordingly.